File: room-microservice/src/main/java/io/rainett/roommicroservice/model/Room.java
package io.rainett.roommicroservice.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "rooms", uniqueConstraints = {
        @UniqueConstraint(name = "room_number_uq", columnNames = "number")
})
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Short number;

    @Column(nullable = false)
    private Short floor;

    @Column(nullable = false)
    private Short positionOnFloor;

    @Enumerated(EnumType.STRING)
    private RoomType roomType;

    @Enumerated(EnumType.STRING)
    private RoomState roomState;

    @ElementCollection
    private List<Long> residentsIds;

    private Byte residentsNumber;

    private BigDecimal price;

    private LocalDateTime createdAt = LocalDateTime.now();

}

File: frontend/angular-app/src/app/room.ts

export interface Room {
  id: number;
  number?: number;
  floor?: number;
  positionOnFloor?: number;
  roomType?: RoomType;
  roomState?: RoomState;
  residentsIds: number[];
  residentsNumber: number;
  price: number;
  createdAt: Date;
}

export enum RoomType {
  SUITE, DELUXE, STANDARD
}

export enum RoomState {
  VACANT, OCCUPIED, RESERVED, SERVICE
}
